enhancements/kube-apiserver/stability-api-removal-notifications.md

```diff
@@ -0,0 +1,159 @@
+
+## Summary
+
+Notify customers that an API that will be removed in the next release is in use.
+
+## Motivation
+
+Kubernetes is being more aggressive with regards to the removal of APIs that have been deprecated or pre-release APIs that have reached a dead-end and will never reach GA. Additionally, the ability to re-enable a previously removed API has been removed.
+
+Customers should be notified that an API that will be removed in the next release is in use so that they can take action to avoid failure when upgrading to the next release. 
+
+### Goals
+
+1. Notify customers an API that will be removed in the next release is in use.
+2. Document how to identify the workload that is using the API to be removed.
+
+### Non-Goals
+
+1. Automatically identify the workload using the API to be removed.
+2. Prevent the customer from attempting to upgrade when an API that will be removed in the next release is in use.
+3. Notify customers that an upstream API is not supported on OpenShift.
+
+## Proposal
+
+Fire an `APIRemovedInNextReleaseInUse` alert when an API that will be removed in the the next release in in use.
+
+On EUS releases, also fire an `APIRemovedInNextEUSReleaseInUse` alert when an API that will be removed in the next extended support release is in use.
+
+### User Stories
+
+#### Story 1
+
+As an OpenShift administrator, I need visibility into which APIs that will be removed in the next release are currently in use, so I can take any needed actions to ensure that I can upgrade to the next release.
+
+#### Story 2
+
+As an OpenShift administrator, I need visibility into which APIs that will be removed in the next EUS release are currently in use, so I can take any needed actions to ensure that I can upgrade to next EUS release.
+
+#### Story 3
+
+As an OpenShift administrator, I need visibilitiy into which workload is using an API that will be removed in the next release, so I can take any needed actions to ensure that I can upgrade to the next release. 
+
+### Implementation Details/Notes/Constraints
+
+#### Pre-release Lifecycle Background
+
+Pre-release APIs are now required to specify `prerelease-lifecycle-gen` tags that inform the apiserver to, among other things, maintain the `apiserver_requested_deprecated_apis` metric.
+
+```go
+// +k8s:prerelease-lifecycle-gen:introduced=1.1
+// +k8s:prerelease-lifecycle-gen:deprecated=1.8
+// +k8s:prerelease-lifecycle-gen:removed=1.18
+// +k8s:prerelease-lifecycle-gen:replacement=apps,v1,Deployment
+```
+
+The `apiserver_requested_deprecated_apis` metrics provides the metadata needed by this enhancement to "look into the future" and identify APIs, currently in use, that will be removed in the next release.
+
+In order to provide the metadata needed to look forward to the next EUS release, `prerelease-lifecycle-gen` tags updated between EUS releases will need to be back-ported to the latest EUS release.
```

Comment:
same metric will do?

Reply:
Yes. The `removed_release` label will be enough to distinguish between next release vs next EUS release.